pennylane/measurements/state.py

```diff
@@ -195,6 +207,17 @@ class DensityMatrixMP(StateMP):
     def __init__(self, wires: Wires, id: Optional[str] = None):
         super().__init__(wires=wires, id=id)
 
+    @classmethod
+    def _abstract_eval(
+        cls, n_wires: Optional[int] = None, shots: Optional[int] = None, num_device_wires: int = 0
+    ):
+        if not has_jax:
+            raise ImportError("jax is required for StateMP._abstract_eval")
+        dtype = jax.numpy.complex128 if jax.config.jax_enable_x64 else jax.numpy.complex64
+        n_wires = n_wires or num_device_wires
+        shape = (2**n_wires, 2**n_wires)
+        return jax.core.ShapedArray(shape, dtype)
```

Comment:
Why are we handling a density matrix using `ShapedArray` but state vectors not? :thinking:

Reply:
i.e. why not 
```suggestion
    @classmethod
    def _abstract_eval(
        cls, n_wires: Optional[int] = None, shots: Optional[int] = None, num_device_wires: int = 0
    ):
        if not has_jax:
            raise ImportError("jax is required for StateMP._abstract_eval")
        n_wires = n_wires or num_device_wires
        shape = (2**n_wires, 2**n_wires)
        return shape, complex
```

Reply:
Because I changed my mind about the call signature and forgot to update everywhere...

Reply:
fixed up.